### PR TITLE
python311Packages.matplotlib: 3.8.4 -> 3.9.0; add a testing derivation to passthru.tests

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
On CI:

- The `x86_64-darwin` build timed out way before it reached `matplotlib`.
- The `x86_64-linux` build shows up as "neutral", but it passed most of it so it seems, besides a weird error with `installCheckPhase` at the end of `sage` build (from `passthru.tests`).
- The `aarch64-linux` build succeeded, but the `passthru.tests.withOutdatedFreetype` failed as expected also at https://github.com/NixOS/nixpkgs/pull/316041#issuecomment-2159216527 .
- The `aarch64-darwin` hasn't passed yet.

So personally I think it'd be OK to let this cook in `staging`, but it would be nice to get an approval from the maintainers (@lovek323 and @veprbl) regarding the form of the 3.9.0 update (part of the 1st commit of the PR).

File: pkgs/development/python-modules/matplotlib/default.nix
Comment (veprbl) [inline pkgs/development/python-modules/matplotlib/default.nix:182]: We don't really need to preserve those, especially if the names are not the same and it takes a 4 line comment to explain it.

**Review by veprbl — APPROVED** [review]
Diff looks good

File: pkgs/development/python-modules/matplotlib/default.nix
Comment (doronbehar) [reply pkgs/development/python-modules/matplotlib/default.nix]: > We don't really need to preserve those, especially if the names are not the same and it takes a 4 line comment to explain it.

Hmm yes I guess you are right.. hopefully no body relied upon that config outside of Nixpkgs.

**Pull request merged**
